source/expressions.tex

```diff
@@ -7451,7 +7451,10 @@
 \end{note}
 
 \item
-a conversion from type \cv{}~\tcode{\keyword{void}*} to a pointer-to-object type;
+a conversion
+from a prvalue \tcode{P} of type ``pointer to \cv{}~\tcode{\keyword{void}}''
+to a pointer-to-object type \tcode{T}
```

Comment:
No, we convert "from an expression to a type". The value category of the source expression makes a difference for the conversion.

Reply:
Yes, there's a bug here. I'll make a core issue.

Reply:
CWG2755

Reply:
A non-object type cannot ever be similar to an object type.

Null pointers seem to be a separate feature. Is there evidence that the paper intended to allow null pointer values?

Reply:
The entire bulleted list presents situations when something is NOT a core constant expression.  If the "unless" part is not satisfied, the conversion remains not a core constant expression.

Regarding null pointers, not everything semantically possible necessarily has syntax to express it concisely.  It might very well be an oversight in the paper that null pointers are not handled, but that's a pure add-on feature (that a later paper can add at any time), not an obvious CWG-level defect.

Also note that this pull request is for applying the wording of the paper to the Working Draft, not for discussing the technical merits of the design intent. Please direct your concerns to the EWG reflector.

Reply:
Agreed. Fixed.